libs/components/modals/testing/src/confirm/confirm-harness.ts

```diff
@@ -0,0 +1,124 @@
+import { ComponentHarness, HarnessQuery } from '@angular/cdk/testing';
+import { SkyComponentHarness } from '@skyux/core/testing';
+import { SkyConfirmType } from '@skyux/modals';
+
+import { SkyConfirmButtonHarness } from './confirm-button-harness';
+import { SkyConfirmButtonHarnessFilters } from './confirm-button-harness-filters';
+
+/**
+ * Harness for interacting with a confirm component in tests.
+ */
+export class SkyConfirmHarness extends SkyComponentHarness {
+  public static hostSelector = 'sky-confirm';
+
+  #getMessageEl = this.locatorFor('.sky-confirm-message');
+  #getBodyEl = this.locatorForOptional('.sky-confirm-body');
+  #getButtons = this.locatorForAll(SkyConfirmButtonHarness);
+
+  /**
+   * Gets the message of the confirm component.
+   */
+  public async getMessageText(): Promise<string> {
+    return (await this.#getMessageEl()).text();
+  }
+
+  /**
+   * Gets the body of the confirm component.
+   */
+  public async getBodyText(): Promise<string | undefined> {
+    return (await this.#getBodyEl())?.text();
+  }
+
+  /**
+   * Gets the type of the confirm component.
+   */
+  public async getType(): Promise<SkyConfirmType> {
+    const buttons = await this.#getButtons();
+
+    if (buttons.length === 1 && (await buttons[0].textContent()) === 'OK') {
```

Comment:
What happens if locales change and the text of the button is no longer "OK"? Is there a class or data attribute on the confirm's component that we can reference (e.g., `sky-confirm-type-ok`)? We can talk offline if needed.

Reply:
There is not one but Paul and I debated adding one and he wasn't against it